src/components/CalendarButtonTwoStep/CalendarButtonTwoStep.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import { findDOMNode } from 'react-dom';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+import { START_DATE, END_DATE } from 'react-dates/constants';
+
+import { RangePickerController } from '../Calendar';
+import Button from '../Button';
+import { textKilo } from '../../styles/style-helpers';
+
+class CalendarButton extends Component {
+  state = { startDate: null, endDate: null, focusedInput: null };
+  buttonRef = null;
+
+  _onDatesChange = ({ startDate, endDate }) => {
+    this.setState({ startDate, endDate });
+
+    if (!endDate) {
+      this._onFocusChange(END_DATE);
+    } else {
+      this._onFocusChange(START_DATE);
+    }
+  };
+
+  _onFocusChange = focusedInput => this.setState({ focusedInput });
+
+  _onClear = () =>
+    this.setState({ startDate: null, endDate: null, focusedInput: null });
+
+  _onButtonClick = () =>
+    this.setState(({ focusedInput }) => ({
+      focusedInput: focusedInput !== null ? null : START_DATE
+    }));
+
+  _currentSelection = () => {
+    const { startDate, endDate } = this.state;
+
+    if (!startDate && !endDate) {
+      return 'Dates';
+    }
+
+    return `${this._toDate(startDate)} - ${this._toDate(endDate)}`;
+  };
+
+  _toDate = date => (date ? date.format('MMM DD') : '');
+
+  _onButtonRef = ref => {
+    this.buttonRef = ref;
+  };
+
+  _onOutsideClick = ({ target }) => {
+    if (this.buttonRef) {
+      // TODO: May be implement forwardRef after we upgrade to 16.3 or elementRef
+      // eslint-disable-next-line react/no-find-dom-node
+      const buttonDomNode = findDOMNode(this.buttonRef);
+
+      if (!buttonDomNode.contains(target)) {
+        this._onFocusChange(null);
+      }
+    }
+  };
+
+  _onCommit = () => {
```

Comment:
Is this when someone confirms the dates? Then I'd maybe go with `handleConfirm`. Or maybe `handleApply`, as that's the default button text.

Reply:
Done